pkg/i2gw/providers/common/utils.go

```diff
@@ -17,24 +17,50 @@ limitations under the License.
 package common
 
 import (
+	"context"
 	"fmt"
 	"regexp"
 
 	networkingv1 "k8s.io/api/networking/v1"
 	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/client-go/kubernetes"
 	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
 )
 
+var k8sClientset *kubernetes.Clientset
+
+func GetDefaultIngressClass() (string, error) {
+	ingressClasses, err := k8sClientset.NetworkingV1().IngressClasses().List(context.Background(), metav1.ListOptions{})
```

Comment:
We support 2 modes for reading resources: from the cluster or a local file.
Here it seems that you're always reading from the cluster. We should come up with a way to take into account the user's choice